src/methods/Method_E.cpp

```diff
@@ -0,0 +1,257 @@
+    const fs::path & scriptsLocation,
+    const double & zPos_measureArea)
+{
+    _measureAreaId = boost::lexical_cast<string>(_lineForMethod_E->_id);
+
+    for(int i = 0; i < peddata.GetNumPeds(); i++) {
+        _passLine.push_back(false);
+    }
+
+    _lenLine = sqrt(
+                   pow((_lineForMethod_E->_lineStartX - _lineForMethod_E->_lineEndX), 2) +
+                   pow((_lineForMethod_E->_lineStartY - _lineForMethod_E->_lineEndY), 2)) *
+               CMtoM;
+    // length of line -> delta x for density
+
+    OpenRhoFileMethodE();
+    LOG_INFO("------------------------Analyzing with Method E-----------------------------");
+    LOG_INFO("Line is selected as measurement area (id {})", _measureAreaId);
+
+    for(const auto & [frameNr, ids] : _peds_t) {
+        int frid = frameNr + _minFrame;
+
+        if(!(frid % 100)) {
+            LOG_INFO("frame ID = {}", frid);
+        }
+
+        vector<int> idsInFrame =
+            peddata.GetIndexInFrame(frameNr, _peds_t[frameNr], zPos_measureArea);
+
+        OutputDensityLine(frameNr, idsInFrame);
+    }
+    fclose(_fRho);
+    // output of density done, now output of flow
+    OutputFlow(_fps, _accumPedsPassLine);
+}
+
+void Method_E::OpenRhoFileMethodE()
+{
+    fs::path tmp("_id_" + _measureAreaId + ".dat");
+    tmp = _outputLocation / "Fundamental_Diagram" / "Method_E" /
+          ("rho_" + _trajName.string() + tmp.string());
+
+    string filename = tmp.string();
+
+    if((_fRho = Analysis::CreateFile(filename)) == nullptr) {
+        LOG_WARNING("cannot open file {} to write density data\n", filename);
+        exit(EXIT_FAILURE);
+    }
+    fprintf(_fRho, "#framerate:\t%.2f\n\n#frame\tdensity(m ^ (-1))\n", _fps);
+}
+
+void Method_E::OutputDensityLine(int frame, const vector<int> & ids)
+{
+    int framePassLine = GetNumberPassLine(frame, ids);
+    // framePassLine -> number of pedestrians that pass the line during this frame
+
+    if(!_accumPedsPassLine.size() == 0) {
+        _accumPedsPassLine.push_back(_accumPedsPassLine.back() + framePassLine);
+    } else {
+        // first frame
+        _accumPedsPassLine.push_back(framePassLine);
+    }
+
+    double density = framePassLine / _lenLine;
+    fprintf(_fRho, "%i\t%.3f\n", frame, density);
+}
+
+int Method_E::GetNumberPassLine(int frame, const vector<int> & ids)
+{
+    // returns number of pedestrians that passed the line during this frame
+    int framePassLine = 0;
+    for(auto const i : ids) {
+        bool IspassLine = false;
+        if(frame > _firstFrame[i] && !_passLine[i]) {
+            IspassLine = IsPassLine(
+                _lineForMethod_E->_lineStartX,
+                _lineForMethod_E->_lineStartY,
+                _lineForMethod_E->_lineEndX,
+                _lineForMethod_E->_lineEndY,
+                _xCor(i, frame - 1),
+                _yCor(i, frame - 1),
+                _xCor(i, frame),
+                _yCor(i, frame));
+        }
+        if(IspassLine == true) {
```

Comment:
== true not needed, as `IspassLine` is already a `bool` value
```suggestion
        if(IspassLine) {
```